### PR TITLE
publishing testdata after successful master build

### PR DESCRIPTION
**Pull request by djsly**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Prereq, to remove testdata from the build. we will be publishing the testdata on master build pipeline

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: .pipelines/.vsts-vhd-builder.yaml [hunk 1]
```diff
@@ -104,3 +104,18 @@ stages:
...
+    jobs:
+      - job: Generate_Publish_Testdata
+        timeoutInMinutes: 10
+        steps:
+        - bash: |
+            make generate
+          displayName: Generate Testdata
+        - task: 1ES.PublishPipelineArtifact@1
+          inputs:
+            targetPath: $(Pipeline.Workspace)/pkg/agent/testdata
```
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder.yaml:118]: how would we consume this from unit test workflows? iterate through the most recent runs and pick out the latest one from master?

I was also originally thinking we'd have a new workflow that would only trigger on merge commits to the master branch which would actually upload/update the testdata, as this stands we'd upload testdata as an artifact to every VHD check-in build

Comment (djsly) [reply .pipelines/.vsts-vhd-builder.yaml:118]: it wont be a workflow, I'm working on a new ado pipeline.

Comment (djsly) [reply .pipelines/.vsts-vhd-builder.yaml:118]: what are checkins build ?

Comment (cameronmeissner) [reply .pipelines/.vsts-vhd-builder.yaml:118]: VHD check-in builds, which is defined by `.vsts-vhd-builder.yaml`

Comment (djsly) [reply .pipelines/.vsts-vhd-builder.yaml:118]: I"m not sure where else we should upload the testdata ? why not on the master build after things gets merged ?

Comment (cameronmeissner) [reply .pipelines/.vsts-vhd-builder.yaml:118]: I think builds off the master branch are fine to upload from, though with this change we'd be uploading artifacts to every instance of the checkin build

Comment (cameronmeissner) [reply .pipelines/.vsts-vhd-builder.yaml:118]: That's also why I was wondering how we'd pick out the correct artifacts from the new pipeline

Comment (cameronmeissner) [reply .pipelines/.vsts-vhd-builder.yaml:118]: Ah okay, I guess V2 of DownloadPipelineArtifact will default to latest from master branch: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines

File: .pipelines/.vsts-vhd-builder.yaml [hunk 2]
```diff
@@ -104,3 +104,18 @@ stages:
...
       - template: ./templates/e2e-template.yaml
+  - stage: Generate and Publish Testdata
+    condition: succeeded()
+    jobs:
+      - job: Generate_Publish_Testdata
+        timeoutInMinutes: 10
+        steps:
+        - bash: |
+            make generate
+          displayName: Generate Testdata
```
Comment (cameronmeissner) [inline .pipelines/.vsts-vhd-builder.yaml:115]: I think this is already the case, but probably worth double checking that the checkin build will be automatically triggered whenever we end up changing template content

**Review by cameronmeissner — APPROVED** [review]
will follow-up with a subsequent PR to tune triggers of the new behavior to master branch builds only

File: .pipelines/.vsts-vhd-builder.yaml (hunk 1 above)
Comment (djsly) [reply .pipelines/.vsts-vhd-builder.yaml:118]: correct, you can also select the version. but I was thinking of pulling latest from Master all the time

**Pull request merged**